### ISSUE TITLE
Error running inside std::thread when window-less

### ISSUE THREAD

**Issue by David-Mckee** [stereolabs/zed-examples#6]
[Ubuntu/Jetson TX1] I'm writing a positional tracking logger, so I've modified the example to allow  running from commandline with no TrackingViewer window. However, when running inside a std::thread, I get several errors. If, however, I run the run(); loop normally (not inside std::thread), all is well. (The only changes I made are logging the Pose to a file stream, and a command line argument to toggle window mode). Below are the errors I get when running std::thread(run) windowless :

ZED (Init) >> Depth mode: Performance
ZED (Init) >> Video mode: VGA@60

in cudaError deleteGPU(sl::MEM, sl::uchar1*) : cuda error [29]: driver shutting down.
ZED (Mat) >> WARNING: DispResizer:credibilityMap Mat with GPU memory allocated should be freed before deleting the Camera object to avoid GPU context issues.
in cudaError deleteGPU(sl::MEM, sl::uchar1*) : cuda error [29]: driver shutting down.
ZED (Mat) >> WARNING: DispResizer:depthResized Mat with GPU memory allocated should be freed before deleting the Camera object to avoid GPU context issues.
in cudaError deleteGPU(sl::MEM, sl::uchar1*) : cuda error [29]: driver shutting down.
ZED (Mat) >> WARNING: Rectifier:MxRight Mat with GPU memory allocated should be freed before deleting the Camera object to avoid GPU context issues.
in cudaError deleteGPU(sl::MEM, sl::uchar1*) : cuda error [29]: driver shutting down.
ZED (Mat) >> WARNING: Rectifier:MyRight Mat with GPU memory allocated should be freed before deleting the Camera object to avoid GPU context issues.
in cudaError deleteGPU(sl::MEM, sl::uchar1*) : cuda error [29]: driver shutting down.
ZED (Mat) >> WARNING: Rectifier:MxLeft Mat with GPU memory allocated should be freed before deleting the Camera object to avoid GPU context issues.
in cudaError deleteGPU(sl::MEM, sl::uchar1*) : cuda error [29]: driver shutting down.
ZED (Mat) >> WARNING: Rectifier:MyLeft Mat with GPU memory allocated should be freed before deleting the Camera object to avoid GPU context issues.
in cudaError deleteGPU(sl::MEM, sl::uchar1*) : cuda error [29]: driver shutting down.
ZED (Mat) >> WARNING: in cudaError deleteGPU(sl::MEM, sl::uchar1*) : cuda error [29]: driver shutting down.
in void sl::Mat::free(sl::MEM) : cuda error [29]: driver shutting down.
in void sl::Mat::free(sl::MEM) : cuda error [29]: driver shutting down.
in cudaError deleteGPU(sl::MEM, sl::uchar1*) : cuda error [29]: driver shutting down.
in cudaError deleteGPU(sl::MEM, sl::uchar1*) : cuda error [29]: driver shutting down.
in void sl::Mat::free(sl::MEM) : cuda error [29]: driver shutting down.
in void sl::Mat::free(sl::MEM) : cuda error [29]: driver shutting down.
in cudaError deleteGPU(sl::MEM, sl::uchar1*) : cuda error [29]: driver shutting down.
in cudaError deleteGPU(sl::MEM, sl::uchar1*) : cuda error [29]: driver shutting down.
in void sl::Mat::free(sl::MEM) : cuda error [29]: driver shutting down.
in cudaError deleteGPU(sl::MEM, sl::uchar1*) : cuda error [29]: driver shutting down.
in cudaError deleteGPU(sl::MEM, sl::uchar1*) : cuda error [29]: driver shutting down.
in cudaError deleteGPU(sl::MEM, sl::uchar1*) : cuda error [29]: driver shutting down.
in cudaError deleteGPU(sl::MEM, sl::uchar1*) : cuda error [29]: driver shutting down.
in cudaError deleteGPU(sl::MEM, sl::uchar1*) : cuda error [29]: driver shutting down.
in void sl::Mat::free(sl::MEM) : cuda error [29]: driver shutting down.
in cudaError deleteGPU(sl::MEM, sl::uchar1*) : cuda error [29]: driver shutting down.
in void sl::Mat::free(sl::MEM) : cuda error [29]: driver shutting down.
in void sl::Mat::free(sl::MEM) : cuda error [29]: driver shutting down.
in void sl::Mat::free(sl::MEM) : cuda error [29]: driver shutting down.
in void sl::Mat::free(sl::MEM) : cuda error [29]: driver shutting down.
in void sl::Mat::free(sl::MEM) : cuda error [29]: driver shutting down.
in cudaError deleteGPU(sl::MEM, sl::uchar1*) : cuda error [29]: driver shutting down.
terminate called without an active exception
Aborted
------------------------------------------------------------------------------------------------------------
I have a bool Xwin that determines whether TrackingViewer Viewer is initialized and used or not. If Xwin = false (default), then the viewer window is not initialized or used. However, these lines cause the above errors in that situation: 
zed_callback = std::thread(run); [inside void startZED()]

If I call run() normally inside startZED() when windowless, its fine. By Windowless, I mean I have a command line argument of "-w" that toggles a bool Xwin = true, otherwise Xwin = false (by default). Then, for the following and related lines I do like so:
if(Xwin) viewer.init(); [inside int main()]
. . .
if(Xwin) viewer.updateZEDPosition(camera_pose.pose_data); [inside void run()]
. . . and so on.

So, basically, if viewer is never used then std::thread(run);  fails.
